server.R

```diff
@@ -91,541 +85,594 @@ server <- function(input, output, session) {
+          },
+          sanitize.text.function = function(x) x,
+          include.colnames = FALSE
+        )
 
+        # UI blocks (result dependent) ---------------------------------------------------------------------------------
 
-      if (advisory_tests != 0) {
-        output$advisory_box <- renderUI({
-          advisory_results_box(
-            message = "num_advisory_tests",
-            table = "table_advisory_tests"
-          )
-        })
-      }
+        if (failed_tests != 0) {
+          output$failed_box <- renderUI({
+            fail_results_box(
+              message = "num_failed_tests",
+              table = "table_failed_tests"
+            )
+          })
+        }
+
+        if (failed_tests == 0) {
+          output$passed_box <- renderUI({
+            pass_results_box()
+          })
+        }
 
-      if (advisory_tests != 0) {
-        output$advisory_box <- renderUI({
-          advisory_results_box(
-            message = "num_advisory_tests",
-            table = "table_advisory_tests"
+        if (failed_tests == 0) {
+          output$passed_box <- renderUI({
+            pass_results_box()
+          })
+        }
+
+        # Fireworks go here
+        if (failed_tests == 0) {
+          show_alert(
+            title = "Success !!",
+            type = "success",
+            tags$span(
+              "Your files can now be uploaded to Explore Education Statistics, see our  ",
+              a(href = "https://rsconnect/rsc/stats-production-guidance/ees.html", "guidance on using EES", target = "_blank"),
+              " for more information."
+            ),
+            html = TRUE
           )
-        })
-      }
+        }
 
-      if (ancillary_tests != 0) {
-        output$ancillary_box <- renderUI({
-          ancillary_box()
-        })
-      }
+        # Dynamic trendy-tabs,
+        if (failed_tests == 0) {
+          shinyjs::show(selector = c(
+            "#trendy_tabs li a[data-value=previewTab]",
+            "#trendy_tabs li a[data-value=obUnitTab]",
+            "#trendy_tabs li a[data-value=indicatorsTab]",
+            "#trendy_tabs li a[data-value=outliersTab]",
+            "#trendy_tabs li a[data-value=geogTab]"
+          ))
+        }
+        else {
+          shinyjs::hide(selector = c(
+            "#trendy_tabs li a[data-value=previewTab]",
+            "#trendy_tabs li a[data-value=obUnitTab]",
+            "#trendy_tabs li a[data-value=indicatorsTab]",
+            "#trendy_tabs li a[data-value=outliersTab]",
+            "#trendy_tabs li a[data-value=geogTab]"
+          ))
+        }
 
+        if (data$mainFile %>% select(geographic_level) %>% distinct() %>% nrow() == 1) {
+          shinyjs::hide(selector = c(
+            "#trendy_tabs li a[data-value=geogTab]"
+          ))
+        }
+        else {
+          shinyjs::show(selector = c(
+            "#trendy_tabs li a[data-value=geogTab]"
+          ))
+        }
```

Comment:
I think this if this was edited to take into account number of tests failing too it will fix the issue with this tab appearing even if the file has failed. Happy for this to be done via a separate PR if easier.

Reply:
I highlighted more lines of code than this but there must be a limit! It's that area, but the if is a few lines above

Reply:
I think there's changes coming to this via sarah's PR, so I'd ignore this for now, this was already there and just tidied with the code styling.